features/shared/checkout_previous_branch_after_success/deleted_current_branch/previous_branch_same.feature

```diff
@@ -0,0 +1,36 @@
+Feature: Allow checking out previous git branch to work correctly after running a Git Town commmand that deletes the current branch
+
+  (see ../same_current_branch/previous_branch_same.feature)
+
+
+  Scenario: checkout previous branch after git-kill deletes current branch
+    Given I have feature branches named "previous" and "current"
+    And I am on the "previous" branch
+    And I switch to the "current" branch
+    And I run `git kill`
+    When I checkout my previous git branch
+    Then I end up on the "previous" branch
+
+
+  Scenario: checkout previous branch after a git-prune-branches deletes current branch
+    Given I have feature branches named "previous" and "current"
+    And the following commit exists in my repository
+      | BRANCH   | LOCATION | FILE NAME     | FILE CONTENT     |
+      | previous | local    | previous_file | previous content |
+    And I am on the "previous" branch
+    And I switch to the "current" branch
+    And I run `git prune-branches`
+    When I checkout my previous git branch
+    Then I end up on the "previous" branch
+
+
+  Scenario: checkout previous branch after git-ship deletes current branch
+    Given I have feature branches named "previous", "current"
+    And the following commit exists in my repository
+      | BRANCH  | LOCATION | FILE NAME    | FILE CONTENT    |
+      | current | remote   | feature_file | feature content |
+    And I am on the "previous" branch
+    And I switch to the "current" branch
+    And I run `git ship -m "feature done"`
+    When I checkout my previous git branch
+    Then I end up on the "previous" branch
```

Comment:
These scenarios seem to all to the same thing. You could either do a Scenario Outline, or - since you need to make sure that each command works - integrate these specs into the most appropriate normal Git Town command spec.